src/charm.py

```diff
@@ -159,6 +162,18 @@ def _validate(self):
         if not self._state.server_status == "ready":
             raise ValueError("ui:temporal relation: server is not ready")
 
+        if not int(self.config["port"]):
```

Comment:
If we're going to do this check here you should probably make a wrapper `get_port` and use that when you initialise the nginx relation above too.